Automattic/wp-calypso
# Site Editor: Prevent opening the navigation sidebar, navigate home instead

#### Changes proposed in this Pull Request

* Add a new `wpcom-site-editor` package to ETK, which will contain all the customizations for the site editor.
* Inject a new `CustomNavigationToggleButton` into the Site Editor that simply navigates back home instead of toggling the navigation sidebar.
* Hide the "Browse all templates" button in the document settings dropdown.

This is a temporary measure for the Dotcom FSE Beta to let the user take advantage of most of the Site Editor features, without the navigation sidebar's complexity hindering their efforts.

#### Testing instructions

This needs to be tested in both local and Calypso, to make sure the redirection lands on the appropriate URL according to the current environment.

* Have two sites (local and Dotcom) ready with the Core Site Editor (start [here](https://horizon.wordpress.com/new?flags=gutenboarding/site-editor) to create a new one on Dotcom).
* `cd apps/editing-toolkit && yarn dev --sync`
* On the local site:
  * Open the Site Editor.
  * Click on the navigation toggle button (upper left corner).
  * Make sure the navigation sidebar doesn't open.
  * Check that you are redirected to the wp-admin dashboard.
* On the Dotcom site:
  * Sandbox the site.
  * Open the Site Editor.
  * Click on the navigation toggle button (upper left corner).
  * Make sure the navigation sidebar doesn't open.
  * Check that you are redirected to the Calypso My Home page for the current site.

Related to #54460
